### PR TITLE
2903 projects list

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to ticket

https://leantime.itkdev.dk/dashboard/show#/tickets/showTicket/2903

#### Description

- Add patch for missing menu active trail on views
- Add menu item for projects in view instead of as content
- Add better exposed filters
- Style projects view (add a bunch of twigs)

#### Screenshot of the result

Screenshot 2024-11-06 at 13 45 46

Screenshot 2024-11-06 at 14 24 13

### CODE REVIEW

File: web/themes/custom/itkdev/itkdev_project_theme/templates/components/tablesort-indicator.html.twig [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <span class="visually-hidden">
+      {{ 'Sort ascending'|t }}
+      </span>
+    {% else -%}
+        <i class="fa-solid fa-sort-down fa-sm"></i>
+      <span class="visually-hidden">
+      {{ 'Sort descending'|t }}
+      </span>
+    {% endif %}
+
```
Comment (sinejespersen) [inline web/themes/custom/itkdev/itkdev_project_theme/templates/components/tablesort-indicator.html.twig:30]:
```suggestion
```

File: docs/Development.md [hunk 2]
```diff
@@ -97,7 +97,7 @@ task drush -- php:eval "Drupal::keyValue('development_settings')->setMultiple(['
 task drush -- cache:rebuild
 ```
 
-## Delvelopment command line incantations
+## Development command line incantations
```
Comment (rimi-itk) [inline docs/Development.md:100]: 🎉

File: web/themes/custom/itkdev/itkdev_project_theme/css/project-styles.css [hunk 3]
```diff
@@ -4,6 +4,71 @@
...
+
+/**
+ * @file
+ *
+ * Replaces standard throbber from stable9 with Olivero's.
+ *
+ * Copied and edited from:
+ * core/themes/stable9/css/system/components/autocomplete-loading.module.css
+ *
+ */
```
Comment (rimi-itk) [inline web/themes/custom/itkdev/itkdev_project_theme/css/project-styles.css:31]: Are you copying and editing from somewhere (Olivero) that has already copied and edited from another place (stable9)?

File: web/themes/custom/itkdev/itkdev_project_theme/templates/components/form-element.html.twig [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  ]
+%}
+<div{{ attributes.addClass(classes) }}>
+  {% if label_display in ['before', 'invisible'] %}
+    {{ label }}
+  {% endif %}
+  {% if prefix is not empty %}
+    <span class="field-prefix">{{ prefix }}</span>
+  {% endif %}
+  {% if description_display == 'before' and description.content or type == 'radio' and description.content %}
```
Comment (rimi-itk) [inline web/themes/custom/itkdev/itkdev_project_theme/templates/components/form-element.html.twig:29]: Rewrite to improve readability:

```suggestion
  {% if description.content and (description_display == 'before' or type == 'radio') %}
```

File: web/themes/custom/itkdev/itkdev_project_theme/templates/components/new-project-button.html.twig [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<a href="/node/add/project" class="font-thin bg-green-700 hover:bg-green-600 text-white py-2 px-4 rounded inline-flex items-center" data-drupal-link-system-path="node/add/project">
```
Comment (rimi-itk) [inline web/themes/custom/itkdev/itkdev_project_theme/templates/components/new-project-button.html.twig:1]: Consider using `path` to get the `href` value (like you usually do in Twig):

```suggestion
{% set path = path('node.add', {node_type: 'project'}) %}
<a href="{{ path }}" class="font-thin bg-green-700 hover:bg-green-600 text-white py-2 px-4 rounded inline-flex items-center" data-drupal-link-system-path="{{ path }}">
```

(`drush route | grep node/add` reveales the route name to use)

File: web/themes/custom/itkdev/itkdev_project_theme/templates/views-exposed-form.html.twig [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @see template_preprocess_views_exposed_form()
+ *
+ * @ingroup themeable
+ */
+#}
+{% if q is not empty %}
+  {#
+    This ensures that, if clean URLs are off, the 'q' is added first,
+    as a hidden form element, so that it shows up first in the POST URL.
```
Comment (rimi-itk) [inline web/themes/custom/itkdev/itkdev_project_theme/templates/views-exposed-form.html.twig:17]: Are you sure? If you `POST` you don't add anything to the URL. Do you mean `GET`? You should always use `GET` to get stuff (Drupal breaks this rule in multiple places).

File: web/themes/custom/itkdev/itkdev_project_theme/templates/views-view.html.twig [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  set classes = [
+    dom_id ? 'js-view-dom-id-' ~ dom_id,
+  ]
+%}
+<div{{ attributes.addClass(classes) }}>
+<div class="flex justify-between my-7">
+<h1 class="text-4xl dark:text-white">{{ view.title }}</h1>
+
+    {{ include('@itkdev_project_theme/components/new-project-button.html.twig') }}
+
```
Comment (rimi-itk) [inline web/themes/custom/itkdev/itkdev_project_theme/templates/views-view.html.twig:45]:
```suggestion
    {{ include('@itkdev_project_theme/components/new-project-button.html.twig') }}
```

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Some comments and suggestions added.

File: web/themes/custom/itkdev/itkdev_project_theme/templates/views-exposed-form.html.twig (hunk 6 above)
Comment (sinejespersen) [reply web/themes/custom/itkdev/itkdev_project_theme/templates/views-exposed-form.html.twig:17]: I am not sure. I copy pasted this solely to add the `grid` classes to `<form...`. should I change the comment to `GET`?

File: web/themes/custom/itkdev/itkdev_project_theme/templates/components/new-project-button.html.twig [hunk 8]
```diff
@@ -1,4 +1,6 @@
-<a href="/node/add/project" class="font-thin bg-green-700 hover:bg-green-600 text-white py-2 px-4 rounded inline-flex items-center" data-drupal-link-system-path="node/add/project">
+{% set path = path('node.add', {node_type: 'project'}) %}
+
+<a href={{ path }} class="font-thin bg-green-700 hover:bg-green-600 text-white py-2 px-4 rounded inline-flex items-center" data-drupal-link-system-path="node/add/project">
```
Comment (rimi-itk) [inline web/themes/custom/itkdev/itkdev_project_theme/templates/components/new-project-button.html.twig:3]: What is the `data-drupal-link-system-path` attribute used for? Should it be set using `path`?

```suggestion
<a href={{ path }} class="font-thin bg-green-700 hover:bg-green-600 text-white py-2 px-4 rounded inline-flex items-center" data-drupal-link-system-path="{{ path }}">
```

File: web/themes/custom/itkdev/itkdev_project_theme/templates/views-exposed-form.html.twig (hunk 6 above)
Comment (rimi-itk) [reply web/themes/custom/itkdev/itkdev_project_theme/templates/views-exposed-form.html.twig:17]: Can't (better) exposed filters add the css class names? And make sure that you `GET`?

File: web/themes/custom/itkdev/itkdev_project_theme/css/project-styles.css [hunk 9]
```diff
@@ -4,6 +4,84 @@
...
+
+.ui-autocomplete .ui-menu-item-wrapper.ui-state-active {
+  margin: 0;
+  border: 0;
+  background-color: rgb(241 245 249); /* bg-slate-100 */
+  color: rgb(71 85 105)
+}
+
+/* This css class is added on the projects view (admin/structure/views/view/projects) */
+.projects-view {
```
Comment (sinejespersen) [inline web/themes/custom/itkdev/itkdev_project_theme/css/project-styles.css:79]: image

This looks as so, and will change in a later pr

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**